ls-ecaps.c

```diff
@@ -950,15 +950,33 @@ dvsec_cxl_gpf_port(struct device *d, int where)
 }
 
 static void
-dvsec_cxl_flex_bus(struct device *d, int where, int rev)
+dvsec_cxl_flex_bus(struct device *d, int where, int rev, int len)
 {
   u16 w;
   u32 l, data;
 
-  if (rev < 1)
-  {
-    printf("\t\tRevision %d not supported\n", rev);
-    return;
+  switch (rev) {
+    case 0:
+      if (len!=0x10) {
+        printf("\t\t<Wrong length for Revision %d>\n", rev);
+        return;
+      }
+      break;
+    case 1:
+      if (len!=0x14) {
+        printf("\t\t<Wrong length for Revision %d>\n", rev);
+        return;
+      }
+      break;
+    case 2:
+      if (len!=0x20) {
+        printf("\t\t<Wrong length for Revision %d>\n", rev);
+        return;
+      }
+      break;
+    default:
+      printf("\t\tRevision %d not supported\n", rev);
+      return;
```

Comment:
Revisions should be forward-compatible, right? Then this code disallow forward compatibility with future rev3.

I think that instead of checking length of data at this place at the beginning of the function, length should be checked later, when trying to access data at requested offset (that length is larger than accessing offset).

Reply:
I fixed it.